src/ui/Button/index.tsx

```diff
@@ -12,13 +13,25 @@ export const Button = ({
   style,
   variant = 'primary',
   ...props
-}: ButtonProps): JSX.Element => (
+}: ButtonProps): JSX.Element | null => {
+  const { colors } = useCozyTheme()
+  const [styles, setStyles] = useState<ButtonStyles | undefined>(undefined)
+
+  useEffect(() => {
+    setStyles(computeStyles(colors))
+  }, [colors])
+
+  if (styles === undefined) {
+    return null
+  }
```

Comment:
It feels weird to have to handle that at the component level. We should think about an abstraction to handle that automatically (or use an existing design system)

Reply:
I changed this a bit based on @zatteo suggestion: https://github.com/cozy/cozy-flagship-app/pull/1062#issuecomment-1852154054